src/ResourceStore.php

```diff
@@ -221,6 +221,22 @@ public function save(Resource $resource)
     }
 
     // }}}
+    // {{{ public function delete()
+    
+    public function delete(Resource $resource)
+    {
+        if ($resource->getId() != '') {
```

Comment:
Might make sense to put this into a method on Resource. Something like `isNew()` or `isSaved()`